### PR TITLE
[FIX] Users without the `view-other-user-info` permission can't use the `users.list` endpoint

### PR DESCRIPTION
**Pull request by LucianoPierdona**
## Proposed changes (including videos or screenshots)

This PR fix the query when a normal users access `users.list`

## Issue(s)

Closes: #25728

### CODE REVIEW

**Comment by ggazzo** [comment]
now this is a break change

File: apps/meteor/app/api/server/lib/users.ts [hunk 1]
```diff
@@ -111,6 +112,10 @@ export function getNonEmptyQuery(query: Query): typeof _defaultQuery | (typeof _
 		],
 	};
 
+	if (!canSeeAllUserInfo) {
+		defaultQuery.$or = defaultQuery.$or.filter((or) => !or['emails.address']);
+	}
+
```
Comment (debdutdeb) [inline apps/meteor/app/api/server/lib/users.ts:118]: Can you invert the logic here please? easier to read `if hasPermission then push elem to array` than `if not hasPermission then create new array by filtering out xyz field` kinda over-wording this but I hope you're getting the point 🙈

**Review by debdutdeb — CHANGES_REQUESTED** [review]
The pr description is saying users need the `view-user-administration` permission, not seeing the check there (in the route options you can add `permissionsRequired: ['view-user-administration']`.

File: apps/meteor/app/api/server/lib/users.ts (hunk 1 above)
Comment (LucianoPierdona) [reply apps/meteor/app/api/server/lib/users.ts]: updated!

**Comment by LucianoPierdona** [comment]
> The pr description is saying users need the `view-user-administration` permission, not seeing the check there (in the route options you can add `permissionsRequired: ['view-user-administration']`.

@debdutdeb my bad, I forgot to update the description but this permission is not needed anymore, looks like the endpoint is going to be available for all users. @matheusbsilva137 can confirm.

**Review by debdutdeb — APPROVED** [review]

**Comment by nishant23122000** [comment]
It's working fine.

**Review by ggazzo — APPROVED** [review]

**Pull request merged**